CRM/Contact/DAO/RelationshipVortex.php

```diff
@@ -0,0 +1,328 @@
+
+  /**
+   * Static instance to hold the table name.
+   *
+   * @var string
+   */
+  public static $_tableName = 'civicrm_relationship_vtx';
+
+  /**
+   * Icon associated with this entity.
+   *
+   * @var string
+   */
+  public static $_icon = 'fa-handshake-o';
+
+  /**
+   * Should CiviCRM log any modifications to this table in the civicrm_log table.
+   *
+   * @var bool
+   */
+  public static $_log = FALSE;
+
+  /**
+   * Relationship ID
+   *
+   * @var int
+   */
+  public $id;
+
+  /**
+   * id of the relationship
+   *
+   * @var int
+   */
+  public $relationship_id;
+
+  /**
+   * id of the relationship
+   *
+   * @var int
+   */
+  public $relationship_type_id;
+
+  /**
+   * The vortex record is a permutation of the original relationship record. The orientation indicates whether it is forward (0; A/B) or reverse (1; B/A) relationship.
+   *
+   * @var int
+   */
+  public $orientation;
+
+  /**
+   * id of the first contact
+   *
+   * @var int
+   */
+  public $near_contact_id;
+
+  /**
+   * name for relationship of near_contact to far_contact.
+   *
+   * @var string
+   */
+  public $relation;
+
+  /**
+   * id of the second contact
+   *
+   * @var int
+   */
+  public $far_contact_id;
+
+  /**
+   * Class constructor.
+   */
+  public function __construct() {
+    $this->__table = 'civicrm_relationship_vtx';
+    parent::__construct();
+  }
+
+  /**
+   * Returns localized title of this entity.
+   */
+  public static function getEntityTitle() {
+    return ts('Relationship Vortexs');
+  }
+
+  /**
+   * Returns foreign keys and entity references.
+   *
+   * @return array
+   *   [CRM_Core_Reference_Interface]
+   */
+  public static function getReferenceColumns() {
+    if (!isset(Civi::$statics[__CLASS__]['links'])) {
+      Civi::$statics[__CLASS__]['links'] = static::createReferenceColumns(__CLASS__);
+      Civi::$statics[__CLASS__]['links'][] = new CRM_Core_Reference_Basic(self::getTableName(), 'relationship_id', 'civicrm_relationship', 'id');
+      Civi::$statics[__CLASS__]['links'][] = new CRM_Core_Reference_Basic(self::getTableName(), 'relationship_type_id', 'civicrm_relationship_type', 'id');
+      Civi::$statics[__CLASS__]['links'][] = new CRM_Core_Reference_Basic(self::getTableName(), 'near_contact_id', 'civicrm_contact', 'id');
+      Civi::$statics[__CLASS__]['links'][] = new CRM_Core_Reference_Basic(self::getTableName(), 'far_contact_id', 'civicrm_contact', 'id');
+      CRM_Core_DAO_AllCoreTables::invoke(__CLASS__, 'links_callback', Civi::$statics[__CLASS__]['links']);
+    }
+    return Civi::$statics[__CLASS__]['links'];
+  }
+
+  /**
+   * Returns all the column names of this table
+   *
+   * @return array
+   */
+  public static function &fields() {
+    if (!isset(Civi::$statics[__CLASS__]['fields'])) {
+      Civi::$statics[__CLASS__]['fields'] = [
+        'id' => [
+          'name' => 'id',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Relationship ID'),
```

Comment:
I would love for this table to just not have an `id` column. It's useless and confusing. Especially with this label!

Reply:
Agree, it's silly to have it there. My impression from @eileenmcnaughton (where she mentioned a module with a bespoke table that lacked `id`) was that it's still rather fiddly to have a table without an `id`.

The label is really confusing. Changed it.

Reply:
Note that if you don't have a primary key then replication is impacted